queries-api/src/main/java/io/corbel/lib/queries/parser/JacksonQueryParser.java

```diff
@@ -115,6 +119,11 @@ private QueryNodeImpl getQueryNode(QueryOperator operator, JsonNode nodeField) t
     }
 
     @SuppressWarnings("rawtypes")
+    private QueryLiteral<?> generateObjectQueryLiteral(JsonNode nodeField) throws MalformedJsonQueryException {
+        return new PositionQueryLiteral(jsonParser, nodeField);
```

Comment:
I guess here we assume that all object query literals are PositionQueryLiterals. This is true now that it is the only one, but I is Ok add the matcher to make it more readable. For instance

`Operator(nodeField) match { case $NEAR => new PositionQueryLiteral(...) } `